main/lsp/LSPPreprocessor.cc

```diff
@@ -311,8 +320,8 @@ LSPPreprocessor::canonicalizeEdits(u4 v, unique_ptr<DidOpenTextDocumentParams> o
     if (config->isUriInWorkspace(uri)) {
         string localPath = config->remoteName2Local(uri);
         if (!config->isFileIgnored(localPath)) {
-            auto file = make_shared<core::File>(move(localPath), move(openParams->textDocument->text),
-                                                core::File::Type::Normal, v);
+            auto fileType = getFileType(localPath, config->opts);
+            auto file = make_shared<core::File>(move(localPath), move(openParams->textDocument->text), fileType, v);
```

Comment:
Why does LSP have to know about core::File::Type::Package? This seems like the sort of thing that should be handled by either `Files.cc` or `packager.cc`

Reply:
This didn't feel right to me, but I didn't have a good solution. LSP currently creates files and needs to know their type _before_ `packager.cc` runs.

I almost added a `isPackage()` method to `Files.cc` that checks the filename (rather than add a new file type), but it would also need to know if `opts.stripePackages` is enabled.

I'm open to suggestions!

Reply:
`¯\_(ツ)_/¯` ok. this is fine then